framework/src/headers/ip/v6/icmp/neighbor.rs

```diff
@@ -0,0 +1,118 @@
+use super::{IcmpMessageType, Icmpv6Header};
+use headers::{CalcChecksums, EndOffset, Ipv6VarHeader};
+use std::default::Default;
+use std::fmt;
+use std::marker::PhantomData;
+use std::net::Ipv6Addr;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct Icmpv6Neighbor<T>
```

Comment:
This is a naming thing, but could we just name this `Icmpv6NDP` or maybe just `NDPMessage`?  With `NDP`, the `ICMP` part is implied because that is just now NDP works.

Reply:
Yeah, lets rename the structs in this PR to NDP and drop the icmp. Icmp is already in the package name so its implied